runtime/caches/redis.ts

```diff
@@ -47,13 +39,33 @@ export const caches: CacheStorage = {
   ): Promise<Response | undefined> => {
     throw new Error("Not Implemented");
   },
-  open: (cacheName: string): Promise<Cache> => {
-    if (!redis) {
-      throw new Error(
-        "Redis coult not be used due to the lack of credentials.",
-      );
-    }
-    const requestURLSHA1 = withCacheNamespace(cacheName);
+  open: async (_: string): Promise<Cache> => {
+    const redis = createClient({
+      url: REDIS_URL,
+    }).on("error", (err: Error) => {
+      console.error(err);
+
+      redis.disconnect();
+    }).on("connect", () => {
+      console.debug("[cache]", "connecting");
+    }).on("reconnecting", () => {
+      console.warn("[cache]", "reconnecting");
+    });
+
+    const ensureConnection = () => {
+      if (!redis.isOpen && !redis.isReady) {
+        console.log("[cache]", "not ready");
+
+        redis.connect();
+
+        return false;
+      }
+
+      return true;
```

Comment:
can connect be called multiple times? what would happen?

in case of both redis.isOpen and redis.isReady being false a sync connect() call will make those true? or it will be async set? if the 2nd option is true so you might have multiples `connect` calls

```suggestion
      if (!redis.isOpen && !redis.isReady) {
        console.log("[cache]", "not ready");

        redis.connect();

        return false;
      }

      return true;
```